include/openssl/core_names.h

```diff
@@ -93,18 +93,27 @@ extern "C" {
 #define OSSL_MAC_NAME_KMAC256       "KMAC256"
 
 /* PKEY parameters */
-/* Diffie-Hellman Parameters */
-#define OSSL_PKEY_PARAM_DH_P         "dh-p"
-#define OSSL_PKEY_PARAM_DH_G         "dh-g"
-#define OSSL_PKEY_PARAM_DH_Q         "dh-q"
+/* Diffie-Hellman/DSA Parameters */
+#define OSSL_PKEY_PARAM_FFC_P        "ffc-p"
+#define OSSL_PKEY_PARAM_FFC_G        "ffc-g"
+#define OSSL_PKEY_PARAM_FFC_Q        "ffc-q"
+
 /* Diffie-Hellman Keys */
 #define OSSL_PKEY_PARAM_DH_PUB_KEY   "dh-pub"
```

Comment:
Do we need individual names for pub, priv or could these just be common to all keys?

Reply:
Not for *all* keys.  RSA has n, e and d...

Reply:
you know what I meant :)

Reply:
Not sure.  One could interpret it to mean that all keys could share the same set of parameters, and simply pick and choose between them, i.e. something like this:

``` C
# define OSSL_PKEY_PARAM_PUB_KEY        "pub"
# define OSSL_PKEY_PARAM_PRIV_KEY       "priv"
# define OSSL_PKEY_PARAM_RSA_N          "n"
# define OSSL_PKEY_PARAM_RSA_E          "e"
# define OSSL_PKEY_PARAM_RSA_D          "d"
```

This is similar to how parameter names and macros are treated in the KDF port...

Reply:
I wasn't sure what the conclusion was here, so I've not made any changes as yet.

Reply:
In that case, we should drop `ffc-` too.  I personally do not mind the shorter variants at all, quite the contrary.  If given via, say, `-pkeyopt` in one of the `pkey` sub-commands, I really cannot see a reason to be more verbose than `-pkeyopt hexn:xxxxxxxxxx`, say...

Reply:
BTW, did you notice @slontis' thumbs up on my comment from yesterday?  I would interpret that as a conclusion of sorts...  good enough for me!

Reply:
*Do* note that we're only talking about the strings here.  The macro names are fine, in my opinion.